scikeras/wrappers.py

```diff
@@ -1310,6 +1322,26 @@ def _type_of_target(self, y: np.ndarray) -> str:
             target_type = type_of_target(self.classes_)
         return target_type
 
+    def _compile_model(self, compile_kwargs: Dict[str, Any]) -> None:
+        if compile_kwargs["loss"] == "auto":
+            if len(self.model_.outputs) > 1:
+                raise ValueError(
+                    'Only single-output models are supported with `loss="auto"`'
+                )
+            if self.target_type_ == "binary":
+                compile_kwargs["loss"] = "binary_crossentropy"
+            elif self.target_type_ == "multiclass":
+                if self.model_.outputs[0].shape[1] == 1:
+                    raise ValueError(
+                        "Multi-class targets require the model to have >1 output unit"
+                    )
+                compile_kwargs["loss"] = "sparse_categorical_crossentropy"
+            else:
+                raise ValueError(
+                    f'`loss="auto"` is not supported for tasks of type {self.target_type_}'
```

Comment:
When is this error triggered? Is there a test for it?

Reply:
If the user passes multilabel indicator, multiclass multioutput, etc.

No test yet, I'll put one in tomorrow

Reply:
Why can't that use case be supported?

Reply:
Multiclass-multioutput can't be supported because we have no way of knowing what loss users want for each output, or even how many outputs there are.

multilabel-indicator can't be differentiated from one-hot encoded targets, you'd need to introspect the model architecture.

These are the same reasons why LabelEncoder does not support these use cases, and the same reason why one-hot encoded targets aren't supported if the user doesn't pass an explicit single loss function and that loss function is cce (which is the only way we can know to one-hot encode the targets without introspecting the model).

We could set up some type of sentinel value from LabelEncoder to handle this, but this produces tight coupling, which I do not like

Reply:
I'm most concerned with one-hot encoded targets, or multiclass-indicator. That's an extremely common usage.

Why can't the loss default to `categorical_crossentropy`, and change that default value in special cases?

Reply:
For ease of documentation, I'd almost rather the loss defaulted to `categorical_crossentropy`.

Reply:
If we do that, we can't support non one-hot encoded multiclass targets since at the point they'd have to be one-hot encoded (LabelClassifier) it isn't known if the user is going to to give us a compiled or uncompiled model.

Reply:
We could introduce a `"multiclass-indicator"` target type but it is a bit dangerous because it can be confused with a multilabel indicator where no samples have multiple classes, and they are very different problems.

Reply:
We can certainly improve the error message to tell users exactly what's going on and how to resolve it. Would that help alleviate the concern?

Reply:
There is a test for this error now.

Note that the initialization flow won't even make it this far unless the user is already subclassing and overriding the default target_encoder, since the default implementation (`LabelEncoder`) will raise its own `NotImplementedError` for these same cases. The only case that will not raise an error until this point is one-hot-encoded targets.

For one-hot-encoded targets, the error will look as follows:

```
`loss="auto"` is not supported for tasks of type multilabel-indicator.
Instead, you must explicitly pass a loss function, for example:
  clf = KerasClassifier(..., loss="categorical_crossentropy")
```